### ISSUE TITLE
Embedding layer (Documentation)

### ISSUE THREAD

**Issue by michael135** [fchollet/keras#7488]
Documentation of an Embedding Layer is bit not clear...

Does embedding layer performs word2vec or something different?
If it performs word2vec, I would suggest to mention it in [documentation](https://keras.io/layers/embeddings/))

And to add some relevant reference.
In opinion current reference  [dropout in RNN](https://arxiv.org/abs/1512.05287) is not directly connected to   word2vec.

keras/keras/layers/embeddings.py
https://github.com/fchollet/keras/blob/master/keras/layers/embeddings.py#L11

**Comment by myutwo150** [comment]
Not necessarily word2vec, it's just a lookup table. You can put any real-valued vectors into it. Also I don't think the dropout in RNN has anything to do with word2vec. The input of RNN can be any vectors, not always word embeddings.

**Comment by jerrypaytm** [comment]
I would like to know what Embedding Layer does. There is an issue running the embedding layer with tensorflow in distributed mode. What is the equivalent tensor in TF to achieve similarly?

**Comment by michael135** [comment]
I found the answer for my question which is:

https://stackoverflow.com/questions/38189713/what-is-an-embedding-in-keras
[from the link above]
As far as I know, the Embedding layer is a simple matrix multiplication that transforms words into their corresponding word embeddings.

The weights of the Embedding layer are of the shape (vocabulary_size, embedding_dimension). For each training sample, its input are integers, which represent certain words. The integers are in the range of the vocabulary size. The Embedding layer transforms each integer i into the ith line of the embedding weights matrix.

In order to quickly do this as a matrix multiplication, the input integers are not stored as a list of integers but as a one-hot matrix. Therefore the input shape is (nb_words, vocabulary_size) with one non-zero value per line. If you multiply this by the embedding weights, you get the output in the shape

(nb_words, vocab_size) x (vocab_size, embedding_dim) = (nb_words, embedding_dim)
So with a simple matrix multiplication you transform all the words in a sample into the corresponding word embeddings.
